src/Features/Core/Portable/ExtractMethod/MethodExtractor.cs

```diff
@@ -199,50 +199,26 @@ private ImmutableArray<AbstractFormattingRule> GetFormattingRules(Document docum
 
         private OperationStatus CheckVariableTypes(
             OperationStatus status,
-            AnalyzerResult analyzeResult,
-            CancellationToken cancellationToken)
+            AnalyzerResult analyzeResult)
         {
             var semanticModel = OriginalSelectionResult.SemanticDocument.SemanticModel;
 
-            // sync selection result to same semantic data as analyzeResult
-            var firstToken = OriginalSelectionResult.GetFirstTokenInSelection();
-            var context = firstToken.Parent;
-
-            status = TryCheckVariableType(semanticModel, context, analyzeResult.GetVariablesToMoveIntoMethodDefinition(cancellationToken), status);
-            status = TryCheckVariableType(semanticModel, context, analyzeResult.GetVariablesToSplitOrMoveIntoMethodDefinition(cancellationToken), status);
-            status = TryCheckVariableType(semanticModel, context, analyzeResult.MethodParameters, status);
-            status = TryCheckVariableType(semanticModel, context, analyzeResult.GetVariablesToMoveOutToCallSite(cancellationToken), status);
-            status = TryCheckVariableType(semanticModel, context, analyzeResult.GetVariablesToSplitOrMoveOutToCallSite(cancellationToken), status);
```

Comment:
this code was incredibly wacky.  it seemed to be trying to slice the analyzeResult.Variables into all abutting subsets, to then check to see if particular disallowed types were used.  But it could just access analyzeResult.Variables insetad.